lib/socrata.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'soda/client'
+
+# This module holds the logic used in lib/tasks/socrata.rake
+module Socrata
+  class << self
+    # Returns both the file name (with timestamp) and the exported data as a String.
+    def export(format)
+      log(:info, 'Exporting dataset...')
+
+      file_format = format || 'CSV'
+      export_data = Decidim::Exporters.find_exporter(file_format).new(collection, serializer).export
+      file_name = export_data.filename('socrata_open_data')
+
+      log(:info, "File created: #{file_name}")
+
+      [file_name, export_data.read]
+    rescue StandardError => error
+      log(:error, error)
+    end
+
+    # Returns a Hashie::Mash that represents the body of the HTTP response.
+    def publish
+      log(:info, 'Pushing data to Socrata...')
+
+      client = SODA::Client.new(credentials)
+      payload = collection.map { |resource| serializer.new(resource).serialize.compact }
+      response = client.post(ENV['SODA_DATASET_IDENTIFIER'], payload)
+
+      log(:info, response.body)
+    rescue StandardError => error
+      log(:error, error)
+    end
+
+    private
+
+    def log(level, message)
+      @logger ||= Logger.new("#{Rails.root}/log/socrata.log")
+      @logger.send(level, message)
+      puts message
+    end
+
+    def collection
+      Decidim::ParticipatoryProcess.where(organization: Decidim::Organization.first)
+    end
+
+    def serializer
+      Decidim::ParticipatoryProcesses::ParticipatoryProcessSerializer
+    end
+
+    def credentials
+      {
+        domain: ENV['SODA_DOMAIN'],
+        username: ENV['SODA_USERNAME'],
+        password: ENV['SODA_PASSWORD'],
+        app_token: ENV['SODA_APP_TOKEN']
```

Comment:
I would configure these credentials in `secrets.yml` and then use them here as `Rails.application.secrets.soda_domain`

Reply:
@tramuntanal, what do you think? He's talking about doing in [like this](https://github.com/gencat/participa/blob/65a093d2a08b9b2d7fd3dbd2b9b71de9e44a019b/config/environments/production.rb#L85)

Reply:
I agree with @isaacmg410 that it is how all external variables are setted in Decidim apps. Yes, let's also set them in secrets.yml to keep doing it the same way.

Reply:
Done. Thanks!

Reply:
Done. Thanks